lib/reporting/reports/enterprise_fee_summary/enterprise_fees_with_tax_report_by_producer.rb

```diff
@@ -88,7 +125,10 @@ def join_tax_rate
         def join_supplier
           proc do |item|
             order = item[:order]
-            order.line_items.map(&:supplier_id).map do |supplier_id|
+            order.line_items
+              .map(&:supplier_id)
+              .filter(&method(:supplier_ids_filter))
```

Comment:
I can't help but wonder if we could filter in the main `LineItems` query, which I imagine to be much  more efficient. But I don't know if that would work or not anyway ;)

Reply:
Today I learned about `&method()`, that's really handy thanks!

Reply:
@dacook 
In `query_result`, I'm trying to do what looks like a left join between the tables:

`Orders` x `Enterprise Fees` x `Tax Rates` x `Suppliers`

the result will be an array of hashes 
```ruby
[entreprise_rate, tax_rate, supplier_id, distributor_id and order_cycle_id] => order
```

Now, if you check the initial set of `orders`, you'll find that's calculated from the `line items`
```ruby
orders = report_line_items.list.map(&:order).uniq
```
Those line items are already filtered by `supplier_id` (as you suggested on your comment).

Reply:
Now, If I don't filter again on this method. The line items that were filtered on the main query will appear again : (
```ruby
            order.line_items
              .map(&:supplier_id)
```
this is why I added `.filter(&method(:supplier_ids_filter))`.